### ISSUE TITLE
Minimal Requirements

### ISSUE THREAD

**Issue by whistle-dev** [luczito/ScooterSharing#18]
- [ ] The use of RecyclerView to show a list of available Scooters near the user or a list of previous Rides from that specific user.
- [ ] Implement a custom Adapter to populate a collection into a ViewHolder using a ViewBinding component.
- [ ] The use of Dialogs to confirm the application’s critical user actions.
- [ ] Your project is organized based on some design pattern (e.g., Model-View-Controller, Model-View-ViewModel, Model-View-Presenter), categories, or application features.
- [ ] The use of an authentication system to authenticate users to your app (including Sign In, Sign Out, and Sign Up modules).
- [ ] The implementation of at least one software test (JVM test and/or instrumented test) to verify the standalone behavior of a unit of code within your app.
- [ ] Draw an entity-relationship diagram to show the structure of your database.
- [ ] Use a local or remote database to store the application data.
- [ ] Store application data (e.g., images) in a public bucket, and use these data in your app.
- [ ] Handle foreground location permissions (when the user requests that your app access device location while your app is in use) in dialogs.
- [ ] Implement the Fused Location Provider API client as an Android Service.
- [ ] Show the current user’s location and available scooters in Google Maps.
- [ ] Create a dataset with QR Codes for all scooters available in your ScooterSharing app.
- [ ] You must share this dataset in the final report as an appendix.
- [ ] Implement a module to read a QR Code before starting a new ride.
- [ ] The user must take a photo of the parked scooter and share it to help the next user to find the scooter on the streets.
